src/main/java/school/lemon/changerequest/java/banking/impl/AccountTransactionManager.java

```diff
@@ -11,25 +11,60 @@
      * @throws NullPointerException if {@code bankAccount == null}
      */
     public AccountTransactionManager(TransactionalBankAccount bankAccount) throws NullPointerException {
-        //TODO: implement me
+        if (bankAccount == null)
+            throw new NullPointerException("bankAccount == null");
+        this.bankAccount = bankAccount;
     }
 
     /**
      * Execute bunch of operations.
+     *
      * @param operations array of account operations
      */
     public void execute(BankOperation[] operations) {
-        //TODO: implement me
+        for (BankOperation operation : operations) {
+            switch (operation.getType()) {
+                case ADD_INTEREST: {
+                    bankAccount.addInterest();
+                    break;
+                }
+                case DEPOSIT: {
+                    bankAccount.deposit(operation.getValue());
+                    break;
+                }
+                case SET_RATE: {
+                    bankAccount.setRate(operation.getValue());
+                    break;
+                }
+                case WITHDRAW: {
+                    bankAccount.withdraw(operation.getValue());
+                    break;
+                }
+            }
+        }
     }
 
     /**
      * Execute bunch of operations in one transaction.
      * In case of any exception all previous operations should be reverted and original exception should be thrown.
      * {@code autoCommit} should be enabled in any case.
+     *
      * @param operations array of account operations
      */
     public void executeInTransaction(BankOperation[] operations) throws IllegalArgumentException {
-        //TODO: implement me
+
+        try {
+            bankAccount.commit();
```

Comment:
Can I add new methods to the interface ? I need <code>get</code> and <code>set</code> methods for <code>latestCommit</code>. 
My idea is to copy <code>latestCommit</code>, then commit changes, try to do operations and revert them in case of an exception and afterall recover the initial state of <code>latestCommit</code>